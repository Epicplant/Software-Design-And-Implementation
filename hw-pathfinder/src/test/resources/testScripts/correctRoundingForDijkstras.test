# Tests that the weighted paths are rounded correctly when their sum is added up

# Creates a new graph
CreateGraph newGraph

# Adds node A to the graph
AddNode newGraph A

# Adds node B to the graph
AddNode newGraph B

# Adds node C to the graph
AddNode newGraph C

# Adds edge of weight 2.72341 to the graph
AddEdge newGraph A B 2.72341

# Adds edge of weight 1.11114 to the graph
AddEdge newGraph B C 1.11114

# Finds the shortest path from node A to node B
FindPath newGraph A B

# Finds the shortest path from node A to node C
FindPath newGraph A C