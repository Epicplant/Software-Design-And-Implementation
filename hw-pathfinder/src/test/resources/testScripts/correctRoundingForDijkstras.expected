# Tests that the weighted paths are rounded correctly when their sum is added up

# Creates a new graph
created graph newGraph

# Adds node A to the graph
added node A to newGraph

# Adds node B to the graph
added node B to newGraph

# Adds node C to the graph
added node C to newGraph

# Adds edge of weight 2.72341 to the graph
added edge 2.723 from A to B in newGraph

# Adds edge of weight 1.11114 to the graph
added edge 1.111 from B to C in newGraph

# Finds the shortest path from node A to node B
path from A to B:
A to B with weight 2.723
total cost: 2.723

# Finds the shortest path from node A to node C
path from A to C:
A to B with weight 2.723
B to C with weight 1.111
total cost: 3.835